cmds/core/wget/wget.go

```diff
@@ -45,16 +46,16 @@ func usage() {
 	os.Exit(2)
 }
 
-func run() (reterr error) {
+func run(args []string) (reterr error) {
 	log.SetPrefix("wget: ")
 
-	if flag.Parse(); flag.NArg() != 1 {
+	if len(args) != 1 {
```

Comment:
if len(args) has to be 1, I think this should be run(url string) error
Otherwise, we are letting coverage force us into writing strange code. It's pretty nonintuitive (to me) to have an array as a parameter that can only be one element.
Your other option is to call run(flag.Arg(0)) in main (https://pkg.go.dev/flag#Arg) which returns "" if the numbered arg does not exist.
If there are no args, flag.Arg(0) will be "", and that will fall through to your error message test on line 57.

Reply:
Sure, this make sense. flag.Args(0) will look much better.